src/main/java/com/github/manolo8/darkbot/utils/login/LoginUtils.java

```diff
@@ -113,22 +114,44 @@ public static void usernameLogin(LoginData loginData) {
         try {
             usernameLogin(loginData, "www");
         } catch (Exception e) {
-            usernameLogin(loginData, "lp");
+            try {
+                usernameLogin(loginData, "lp");
+            } catch (IOException ex) {
+                throw new RuntimeException(ex);
+            }
         }
     }
 
-    public static void usernameLogin(LoginData loginData, String domain) {
-        String loginUrl = Http.create("https://" + domain + ".darkorbit.com/")
-                .consumeInputStream(LoginUtils::getLoginUrl);
+    private static void usernameLogin(LoginData loginData, String domain) throws IOException {
+        URL url = new URL("https://" + domain + ".darkorbit.com/");
+        String frontPage = IOUtils.read(Http.create(url.toString()).getInputStream());
+
+        Map<String, String> extraPostParams = Collections.emptyMap();
+        CaptchaAPI solver = CaptchaAPI.getInstance();
+        if (solver != null) {
+            try {
+                extraPostParams = solver.solveCaptcha(url, frontPage);
+            } catch (Exception e) {
+                System.out.println("Captcha solver failed to resolve login captcha");
+                e.printStackTrace();
+            }
+        }
+
+        String loginUrl = getLoginUrl(frontPage);
 
         CookieManager cookieManager = new CookieManager();
         CookieHandler.setDefault(cookieManager);
 
         try {
-            Http.create(loginUrl, Method.POST)
-                    .setParam("username", loginData.getUsername())
-                    .setParam("password", loginData.getPassword())
-                    .closeInputStream();
+            Http http = Http.create(loginUrl, Method.POST)
+                        .setParam("username", loginData.getUsername())
+                        .setParam("password", loginData.getPassword());
+            extraPostParams.forEach(http::setParam);
+            if(http.getContent().contains("bgcdw_errors_all_wrapper")){
+                System.out.println("Wrong Credentials");
+                throw new WrongCredentialsException();
+            }
+
```

Comment:
```suggestion
            if (http.getContent().contains("bgcdw_errors_all_wrapper"))
                throw new WrongCredentialsException();
```

Although i think this is simply handled lower down by the fact that no DOSID cookie will be found, so i'm not sure that this makes much of a difference

Reply:
yup you are right i didnt see it was being handled